src/Type/Definition/ResolveInfo.php

```diff
@@ -130,6 +131,12 @@ private function foldSelectionSet(SelectionSet $selectionSet, $descend)
                     $fragment = $this->fragments[$spreadName];
                     $fields += $this->foldSelectionSet($fragment->selectionSet, $descend);
                 }
+            } else if ($selectionAST instanceof InlineFragment) {
+                $fragmentFields = $this->foldSelectionSet($selectionAST->selectionSet, $descend - 1);
+                $fragmentType = $selectionAST->typeCondition->name->value;
+                foreach ($fragmentFields as $name => $subFields) {
+                    $fields[$fragmentType . ':' . $name] = $subFields;
```

Comment:
Not really. The type condition is important, because implementor A and B could both query field X, but have different children, because they mean something else:

```
... on A {
  some {
    type
  }
}
... on B {
  some {
    kind
  }
}
```

Would B win and overwrite A's `type`?